src/libstd/process.rs

```diff
@@ -0,0 +1,835 @@
+        self.stderr = Some(cfg);
+        self
+    }
+
+    fn spawn_inner(&self, default_io: Stdio) -> IoResult<Child> {
+        let (their_stdin, our_stdin) = try!(
+            setup_io(self.stdin.as_ref().unwrap_or(&default_io), 0, true)
+        );
+        let (their_stdout, our_stdout) = try!(
+            setup_io(self.stdout.as_ref().unwrap_or(&default_io), 1, false)
+        );
+        let (their_stderr, our_stderr) = try!(
+            setup_io(self.stderr.as_ref().unwrap_or(&default_io), 2, false)
+        );
+
+        match ProcessImp::spawn(&self.inner, their_stdin, their_stdout, their_stderr) {
+            Err(e) => Err(e),
+            Ok(handle) => Ok(Child {
+                handle: handle,
+                status: None,
+                stdin: our_stdin.map(|fd| ChildStdin { inner: fd }),
+                stdout: our_stdout.map(|fd| ChildStdout { inner: fd }),
+                stderr: our_stderr.map(|fd| ChildStderr { inner: fd }),
+            })
+        }
+    }
+
+    /// Executes the command as a child process, returning a handle to it.
+    pub fn spawn(&self) -> IoResult<Child> {
+        self.spawn_inner(Stdio::Inherit)
+    }
+
+    /// Executes the command as a child process, waiting for it to finish and
+    /// collecting all of its output.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use std::old_io::Command;
+    ///
+    /// let output = match Command::new("cat").arg("foot.txt").output() {
+    ///     Ok(output) => output,
+    ///     Err(e) => panic!("failed to execute process: {}", e),
+    /// };
+    ///
+    /// println!("status: {}", output.status);
+    /// println!("stdout: {}", String::from_utf8_lossy(output.output.as_slice()));
+    /// println!("stderr: {}", String::from_utf8_lossy(output.error.as_slice()));
+    /// ```
+    pub fn output(&self) -> IoResult<Output> {
+        self.spawn_inner(Stdio::Capture).and_then(|p| p.wait_with_output())
+    }
+
+    /// Executes a command as a child process, waiting for it to finish and
+    /// collecting its exit status.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use std::old_io::Command;
+    ///
+    /// let status = match Command::new("ls").status() {
+    ///     Ok(status) => status,
+    ///     Err(e) => panic!("failed to execute process: {}", e),
+    /// };
+    ///
+    /// println!("process exited with: {}", status);
+    /// ```
+    pub fn status(&self) -> IoResult<ExitStatus> {
+        self.spawn().and_then(|mut p| p.wait())
+    }
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl fmt::Debug for Command {
+    /// Format the program and arguments of a Command for display. Any
+    /// non-utf8 data is lossily converted using the utf8 replacement
+    /// character.
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        try!(write!(f, "{:?}", self.inner.program));
+        for arg in &self.inner.args {
+            try!(write!(f, " '{:?}'", arg));
```

Comment:
One thing I've noticed recently is that the surrounding single-ticks are actually duplicating the quotes inserted by `:?`, so I think they can be removed.